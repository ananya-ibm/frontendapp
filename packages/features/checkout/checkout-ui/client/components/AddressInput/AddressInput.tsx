/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/
import React from 'react';
import { useIntl } from '@exo/frontend-common-i18n';
import { LayoutSpacing } from '@exo/frontend-components-core';
import { AddressForm as ExoAddressForm } from '@exo/frontend-components-forms';
import { removeNull } from '@exo/frontend-common-utils';
import { validatorFactory } from '@exo/frontend-common-validator-factory';
import * as yup from 'yup';
import { Address } from '@exo/frontend-features-checkout-logic';
import { AddressSearch } from '@exo/frontend-features-account-address-search';
import { Button, ButtonGroup } from '@exo/frontend-components-base';
import * as S from './AddressInput.styles';
import { BillingAddressEnableCheckbox } from '../BillingAddressEnableCheckbox/BillingAddressEnableCheckbox';

export const AddressInput = ({
  countriesList,
  isBillingEnabled,
  onToggleUseBillingAddress,
  addressSearch,
  addressToDisplay,
  onFormSubmit = () => {},
  onFormBack = () => {},
  type = 'billing'
}: Props) => {
  const intl = useIntl('features.checkout.checkout-ui.components.AddressInput');

  // TODO: Remove once "Exo"AddressForm supports countryCode
  const fixedAddress = removeNull({ ...addressToDisplay, country: addressToDisplay?.countryCode });

  return (
    <div>
      <LayoutSpacing size="sm" />

      <ExoAddressForm
        intlPrefix="features.checkout.checkout-ui.components.AddressInput"
        dataId={addressToDisplay?.id}
        type={type}
        onSubmit={v => {
          const values = v;
          values.province = !values.province || values.province === '' ? 'N/A' : values.province;
          values.id = addressToDisplay?.id;
          onFormSubmit(values);
        }}
        onCancel={onFormBack}
        data={fixedAddress}
        countries={countriesList}
        fields={[
          ...(addressSearch ? ['addressSearch'] : []),
          ...(type === 'shipping' ? ['phone', 'email'] : []),
          'firstName',
          'lastName',
          'address1',
          'address2',
          'country',
          'city',
          'province',
          'zip'
        ]}
        schema={yup
          .object()
          .shape(
            validatorFactory(
              [
                ...(type === 'shipping' ? ['phone', 'email*'] : []),
                'firstName*',
                'lastName*',
                'address1*',
                'province',
                'city*',
                'country*',
                'zip*'
              ],
              true
            )
          )}
        addressSearch={addressSearch}
        cancelLabel={intl.msg('exoaddressform.back', 'Back')}
        saveLabel={intl.msg('exoaddressform.continue', 'Continue')}
        renderFooter={() => (
          <>
            <S.Footer>
              {onToggleUseBillingAddress && (
                <S.BillingCheckbox>
                  <BillingAddressEnableCheckbox
                    isBillingEnabled={isBillingEnabled ?? false}
                    onCheck={() => onToggleUseBillingAddress!()}
                  />
                </S.BillingCheckbox>
              )}
              <S.Buttons>
                <ButtonGroup isLeft>
                  <Button
                    variant="link"
                    onClick={onFormBack}
                    label={intl.msg('exoaddressform.back', 'Back')}
                  />
                  <Button
                    variant="tertiary"
                    type="submit"
                    label={intl.msg('exoaddressform.continue', 'Continue')}
                  />
                </ButtonGroup>
              </S.Buttons>
            </S.Footer>
          </>
        )}
      />

      <LayoutSpacing size="sm" />
    </div>
  );
};

AddressInput.Skeleton = () => <div>Loading...</div>;

type Props = {
  type?: 'shipping' | 'billing';
  addressToDisplay?: Address;
  addressSearch?: AddressSearch;
  isBillingEnabled?: boolean;
  onToggleUseBillingAddress?: () => void;
  onFormSubmit?: (addr: Address) => void;
  onFormBack?: () => void;
  countriesList?: {
    name: string;
    value: string;
    phonePrefix: string;
    states?: {
      name: string;
      value: string;
    }[];
  }[];
};
