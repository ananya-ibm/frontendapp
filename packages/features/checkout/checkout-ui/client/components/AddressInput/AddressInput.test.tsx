/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderWithTheme as render, getStories } from '@testUtils';
import * as stories from './AddressInput.stories';
import { AddressInput } from './AddressInput';

const mockedOnFormBackFunction = jest.fn();

type Props = Parameters<typeof AddressInput>[0];

const testProps = {
  addressToDisplay: {
    firstName: 'I am test 1',
    lastName: 'I am test 2'
  },
  type: 'billing',
  onFormBack: mockedOnFormBackFunction
} as Props;

describe('<AddressInput /> component', () => {
  test.each(getStories(stories))('its snapshot matches story %s', async (_name, Story) => {
    const { container } = render(<Story {...Story.args} />);
    expect(container).toMatchSnapshot();
  });
  test('on pressing "Back" button, onFormBack is called', () => {
    const { container, getByText } = render(<AddressInput {...testProps} />);
    expect(container).toBeDefined();
    fireEvent.click(getByText('Back'));
    expect(mockedOnFormBackFunction).toHaveBeenCalled();
  });
});
