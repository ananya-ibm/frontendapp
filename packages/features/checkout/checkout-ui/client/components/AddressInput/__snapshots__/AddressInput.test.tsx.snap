// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressInput /> component its snapshot matches story Billing 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  color: #0f62fe;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: bold 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #0043ce;
}

.c9 svg {
  color: #161616;
}

.c8 {
  font-size: 105%;
  font-weight: 700;
  line-height: 0;
  white-space: nowrap;
}

.c8 svg {
  max-height: 100%;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  gap: 1rem;
}

.c0 {
  height: 1rem;
}

.c3 {
  margin-bottom: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.5rem;
}

.c1 > .field-wrapper {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin-left: 1.5rem;
}

.c1 > .field-wrapper:first-child {
  margin-left: 0;
}

.c1 .c2 {
  margin-bottom: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.5rem;
}

.c5 {
  display: block;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 > div {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

@media (min-width:672px) {
  .c7 {
    -webkit-flex-wrap: no-wrap;
    -ms-flex-wrap: no-wrap;
    flex-wrap: no-wrap;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:672px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c1 > .field-wrapper {
    margin-left: 0;
    margin-top: 1.5rem;
  }

  .c1 > .field-wrapper:first-child {
    margin-top: 0;
  }
}

<div>
  <div>
    <div
      class="c0"
    />
    <form>
      <div
        class="FormBodystyles__FormBody-sc-12wq4uu-0"
      >
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_firstName"
              >
                Your first name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_firstName"
                    name="firstName"
                    placeholder="Your first name"
                    required=""
                    title="Your first name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_lastName"
              >
                Your last name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_lastName"
                    name="lastName"
                    placeholder="Your last name"
                    required=""
                    title="Your last name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address1"
            >
              Your address 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address1"
                  name="address1"
                  placeholder="Address"
                  required=""
                  title="Address"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address2"
            >
              Additional address line (optional)
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address2"
                  name="address2"
                  placeholder="Address line 2"
                  title="Address line 2"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_country"
            >
              Country 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_country"
                  name="country"
                  placeholder="Country"
                  required=""
                  title="Country"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_city"
              >
                City 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_city"
                    name="city"
                    placeholder="City"
                    required=""
                    title="City"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_province"
              >
                State/province (optional)
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_province"
                    name="province"
                    placeholder="State/province"
                    title="State/province"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_zip"
              >
                Postal/Zip Code 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_zip"
                    name="zip"
                    placeholder="Postal/Zip Code"
                    required=""
                    title="Postal/Zip Code"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <button
                class="c8 cds--btn cds--btn--tertiary"
                tabindex="0"
                type="submit"
              >
                Continue
                 
              </button>
              <a
                class="c9"
              >
                Back
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      class="c0"
    />
  </div>
</div>
`;

exports[`<AddressInput /> component its snapshot matches story Default 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  color: #0f62fe;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: bold 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #0043ce;
}

.c9 svg {
  color: #161616;
}

.c8 {
  font-size: 105%;
  font-weight: 700;
  line-height: 0;
  white-space: nowrap;
}

.c8 svg {
  max-height: 100%;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  gap: 1rem;
}

.c0 {
  height: 1rem;
}

.c3 {
  margin-bottom: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.5rem;
}

.c1 > .field-wrapper {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin-left: 1.5rem;
}

.c1 > .field-wrapper:first-child {
  margin-left: 0;
}

.c1 .c2 {
  margin-bottom: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.5rem;
}

.c5 {
  display: block;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 > div {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

@media (min-width:672px) {
  .c7 {
    -webkit-flex-wrap: no-wrap;
    -ms-flex-wrap: no-wrap;
    flex-wrap: no-wrap;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:672px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c1 > .field-wrapper {
    margin-left: 0;
    margin-top: 1.5rem;
  }

  .c1 > .field-wrapper:first-child {
    margin-top: 0;
  }
}

<div>
  <div>
    <div
      class="c0"
    />
    <form>
      <div
        class="FormBodystyles__FormBody-sc-12wq4uu-0"
      >
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_firstName"
              >
                Your first name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_firstName"
                    name="firstName"
                    placeholder="Your first name"
                    required=""
                    title="Your first name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_lastName"
              >
                Your last name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_lastName"
                    name="lastName"
                    placeholder="Your last name"
                    required=""
                    title="Your last name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_email"
            >
              Your email address 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_email"
                  name="email"
                  placeholder="Your email address"
                  required=""
                  title="Your email address"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_phone"
            >
              Your phone number (optional)
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_phone"
                  name="phone"
                  placeholder="Your phone number"
                  title="Your phone number"
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address1"
            >
              Your address 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address1"
                  name="address1"
                  placeholder="Address"
                  required=""
                  title="Address"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address2"
            >
              Additional address line (optional)
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address2"
                  name="address2"
                  placeholder="Address line 2"
                  title="Address line 2"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_country"
            >
              Country 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_country"
                  name="country"
                  placeholder="Country"
                  required=""
                  title="Country"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_city"
              >
                City 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_city"
                    name="city"
                    placeholder="City"
                    required=""
                    title="City"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_province"
              >
                State/province (optional)
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_province"
                    name="province"
                    placeholder="State/province"
                    title="State/province"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_zip"
              >
                Postal/Zip Code 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_zip"
                    name="zip"
                    placeholder="Postal/Zip Code"
                    required=""
                    title="Postal/Zip Code"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <button
                class="c8 cds--btn cds--btn--tertiary"
                tabindex="0"
                type="submit"
              >
                Continue
                 
              </button>
              <a
                class="c9"
              >
                Back
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      class="c0"
    />
  </div>
</div>
`;

exports[`<AddressInput /> component its snapshot matches story Skeleton 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;

exports[`<AddressInput /> component its snapshot matches story WithInput 1`] = `
.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  color: #0f62fe;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: bold 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover {
  color: #0043ce;
}

.c9 svg {
  color: #161616;
}

.c8 {
  font-size: 105%;
  font-weight: 700;
  line-height: 0;
  white-space: nowrap;
}

.c8 svg {
  max-height: 100%;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  gap: 1rem;
}

.c0 {
  height: 1rem;
}

.c3 {
  margin-bottom: 1.5rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.5rem;
}

.c1 > .field-wrapper {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin-left: 1.5rem;
}

.c1 > .field-wrapper:first-child {
  margin-left: 0;
}

.c1 .c2 {
  margin-bottom: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.5rem;
}

.c5 {
  display: block;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 > div {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

@media (min-width:672px) {
  .c7 {
    -webkit-flex-wrap: no-wrap;
    -ms-flex-wrap: no-wrap;
    flex-wrap: no-wrap;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:672px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .c1 > .field-wrapper {
    margin-left: 0;
    margin-top: 1.5rem;
  }

  .c1 > .field-wrapper:first-child {
    margin-top: 0;
  }
}

<div>
  <div>
    <div
      class="c0"
    />
    <form>
      <div
        class="FormBodystyles__FormBody-sc-12wq4uu-0"
      >
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_firstName"
              >
                Your first name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_firstName"
                    name="firstName"
                    placeholder="Your first name"
                    required=""
                    title="Your first name"
                    type="text"
                    value="John"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_lastName"
              >
                Your last name 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_lastName"
                    name="lastName"
                    placeholder="Your last name"
                    required=""
                    title="Your last name"
                    type="text"
                    value="Doe"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address1"
            >
              Your address 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address1"
                  name="address1"
                  placeholder="Address"
                  required=""
                  title="Address"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_address2"
            >
              Additional address line (optional)
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_address2"
                  name="address2"
                  placeholder="Address line 2"
                  title="Address line 2"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c2 c3"
        >
          <div
            class="cds--form-item cds--text-input-wrapper"
          >
            <label
              class="cds--label"
              for="address_country"
            >
              Country 
            </label>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  class="cds--text-input cds--text__input cds--text-input--md"
                  id="address_country"
                  name="country"
                  placeholder="Country"
                  required=""
                  title="Country"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c1"
        >
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_city"
              >
                City 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_city"
                    name="city"
                    placeholder="City"
                    required=""
                    title="City"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_province"
              >
                State/province (optional)
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_province"
                    name="province"
                    placeholder="State/province"
                    title="State/province"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="field-wrapper"
            style="flex-basis: 100%;"
          >
            <div
              class="cds--form-item cds--text-input-wrapper"
            >
              <label
                class="cds--label"
                for="address_zip"
              >
                Postal/Zip Code 
              </label>
              <div
                class="cds--text-input__field-outer-wrapper"
              >
                <div
                  class="cds--text-input__field-wrapper"
                >
                  <input
                    class="cds--text-input cds--text__input cds--text-input--md"
                    id="address_zip"
                    name="zip"
                    placeholder="Postal/Zip Code"
                    required=""
                    title="Postal/Zip Code"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <button
                class="c8 cds--btn cds--btn--tertiary"
                tabindex="0"
                type="submit"
              >
                Continue
                 
              </button>
              <a
                class="c9"
              >
                Back
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      class="c0"
    />
  </div>
</div>
`;
